File: asignatura.py
class Asignatura:
	"""
	Representa una asignatura, identificada por su denominación.
	"""
	def __init__(self, denominacion, num_trim):
		self.__set_denom(denominacion)
		self.__set_num_trimestres(num_trim)

	def __repr__(self):
		return f"Asignatura('{self.denom()}', {self.num_trimestres()})"

	def __str__(self):
		return f'Es la asignatura {self.denom()}, que consta de {self.num_trimestres()} trimestre(s).'

	def denom(self):
		return self.__denom

	def num_trimestres(self):
		return self.__num_trimestres

	def __set_denom(self, denominacion):
		self.__denom = denominacion

	def __set_num_trimestres(self, num_trim):
		self.__num_trimestres = num_trim

File: Asignatura.java
/**
 * Representa una asignatura, identificada por su \
 * denominación y nº de trimestres.
 */
public class Asignatura{
  private String denom;
  private int numTrimestres;
  /**
   * Constructor de la clase Asignatura.
   */
  Asignatura(String denom, int numTrim){
		this.setDenom(denom);
		this.setNumTrimestres(numTrim);
    }

  /*
	def __repr__(self):
		"""Devuelve la forma normal de la asignatura."""
		return f"Asignatura('{self.denom()}', {self.num_trimestres()})"
  */

  /**
   * Devuelve un literal de cadena que representa a la asignatura.
   * @return La representación literal del alumno.
   */
	public String toString(){
		return "Es la asignatura " + this.denom() + ", que consta de " + this.numTrimestres() + " trimestre(s).";
    }

  /**
   * Devuelve la denominación de la asignatura.
   * @return La denominación de la asignatura.
   */
	public String denom(){
		return this.denom;
    }

  /**
   * Devuelve el nº de trimestres de la asignatura.
   */
	public int numTrimestres(){
		return this.numTrimestres;
    }

  /**
   * Asigna una denominación a la asignatura.
   * @param denom La nueva denominación.
   */
	private void setDenom(String denom){
		this.denom = denom;
    }

  /**
   * Asigna un nº de trimestres a la asignatura.
   * @param numTrim El nuevo nº de trimestres.
   */
	private void setNumTrimestres(int numTrim){
		this.numTrimestres = numTrim;
    }
}
